pkg/apis/certmanager/validation/issuer.go

```diff
@@ -179,6 +182,23 @@ func ValidateACMEIssuerDNS01Config(iss *v1alpha1.ACMEIssuerDNS01Config, fldPath
 				}
 			}
 		}
+		if p.Execute != nil {
+			if numProviders > 0 {
+				el = append(el, field.Forbidden(fldPath.Child("execute"), "may not specify more than one provider type"))
+			} else {
+				numProviders++
+				sks := &p.Execute.EnvSecret
+				secretFld := fldPath.Child("execute", "envSecretSecretRef")
+				if sks.Name == "" {
```

Comment:
This field needn't be required - some users may not required secrets to be injected.